src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -502,6 +511,9 @@ private void CancellationCallback()
 
                     _wdToPathMap.Clear();
                 }
+
+                // Unlock the root directory
+                Debug.Assert(Interop.Sys.FLock(_inotifyHandle, Interop.Sys.LockOperations.LOCK_UN) == 0);
```

Comment:
Good catch, I'll move this into the finally block and remove the assert since it really isn't actionable